subprojects/docs/src/docs/userguide/multi_project_builds.adoc

```diff
@@ -48,403 +48,660 @@ Configuration on demand mode attempts to configure only projects that are releva
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/multiproject/useSubprojects/water/build.gradle[]
+----
+
+.Output of **`gradle -q hello`**
+----
+> gradle -q hello
+include::{samplesPath}/userguide/multiproject/useSubprojects/water/multiprojectUseSubprojects.out[]
+----
 
 You may notice that there are two code snippets referencing the “`hello`” task. The first one, which uses the “`task`” keyword, constructs the task and provides it's base configuration. The second piece doesn't use the “`task`” keyword, as it is further configuring the existing “`hello`” task. You may only construct a task once in a project, but you may add any number of code blocks providing additional configuration.
 
 [[sub:adding_specific_behavior]]
-==== Adding specific behavior
+=== Adding specific behavior
 
 You can add specific behavior on top of the common behavior. Usually we put the project specific behavior in the build script of the project where we want to apply this specific behavior. But as we have already seen, we don't have to do it this way. We could add project specific behavior for the `bluewhale` project like this:
 
-++++
-<sample id="multiprojectSubprojectsAddFromTop" dir="userguide/multiproject/subprojectsAddFromTop/water" title="Defining specific behaviour for particular project">
-                <sourcefile file="build.gradle"/>
-                <output args="-q hello"/>
-            </sample>
-++++
+=== Example: Defining specific behaviour for particular project
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/userguide/multiproject/subprojectsAddFromTop/water/build.gradle[]
+----
+
+.Output of **`gradle -q hello`**
+----
+> gradle -q hello
+include::{samplesPath}/userguide/multiproject/subprojectsAddFromTop/water/multiprojectSubprojectsAddFromTop.out[]
+----
 
 As we have said, we usually prefer to put project specific behavior into the build script of this project. Let's refactor and also add some project specific behavior to the `krill` project.
 
-++++
-<sample id="multiprojectSpreadSpecifics" dir="userguide/multiproject/spreadSpecifics/water" includeLocation="true" title="Defining specific behaviour for project krill">
-                <layout>
-                    build.gradle
-                    settings.gradle
-                    bluewhale/build.gradle
-                    krill/build.gradle
-                </layout>
-                <sourcefile file="settings.gradle"/>
-                <sourcefile file="bluewhale/build.gradle"/>
-                <sourcefile file="krill/build.gradle"/>
-                <sourcefile file="build.gradle"/>
-                <output args="-q hello"/>
-            </sample>
-++++
+=== Example: Defining specific behaviour for project krill
+
+.Project layout
+----
+.
+├── bluewhale
+│   └── build.gradle
+├── build.gradle
+├── krill
+│   └── build.gradle
+└── settings.gradle
+----
+
+[source.multi-language-sample,groovy]
```

Comment:
:x: Something if off with the rendering of this series of samples, all tabs are grouped above the first sample:

![image](https://user-images.githubusercontent.com/132773/42991145-09d09a42-8c05-11e8-84ce-4cc02b3069b9.png)

Reply:
The multi-language sample JS groups all multi-language samples that don't have anything in between them. I'm going to add some spacers everywhere to prevent this. It would be better to have some kind of macro that allowed one to declare multiple source files for a sample, but I don't want to block this effort on developing that.

Reply:
Scratch that :) TIL that blocks can be nested in Asciidoc. Fixed in b42cd6a9b2dbefbdad7976d973dee8549aae1ab2